src/GitVersionCore/Helpers/LockFile.cs

```diff
@@ -0,0 +1,208 @@
+/*
+MIT License
+
+Copyright 2020 Teroneko
+
+Permission is hereby granted, free of charge, to any person obtaining a copy 
+of this software and associated documentation files (the "Software"), to deal 
+in the Software without restriction, including without limitation the rights to 
+use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies 
+of the Software, and to permit persons to whom the Software is furnished to do 
+so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all 
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR 
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR 
+COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER 
+IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN 
+CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+using System;
+using System.IO;
+using System.Threading;
+
+namespace GitVersion.Helpers
+{
+    /// <summary>
+    /// This helper class can lock files.
+    /// </summary>
+    public static class LockFile
+    {
+        public const FileMode DefaultFileMode = FileMode.OpenOrCreate;
+        public const FileAccess DefaultFileAccess = FileAccess.ReadWrite;
+        public const FileShare DefaultFileShare = FileShare.None;
```

Comment:
Is it `FileShare.None` that is going to ensure that only one process is able to access the specified `filePath`? Can this be guaranteed across all file systems or is this file system dependent?

Reply:
Yes right. that's the key. So there you have your global lock over all processes . 🙂

Reply:
The API needs to add `FileStream.Lock(0,0);`

See references for unix lock support:
https://github.com/dotnet/coreclr/pull/8233/
https://stackoverflow.com/questions/35444470/how-to-lock-unlock-a-file-across-process

Reply:
The API needs to add FileStream.Lock(0,0);

See references for unix lock support:
dotnet/coreclr#8233
https://stackoverflow.com/questions/35444470/how-to-lock-unlock-a-file-across-process